libbeat/dashboards/modify_json.go

```diff
@@ -159,25 +169,33 @@ func ReplaceIndexInVisState(logger *logp.Logger, index string, visStateJSON stri
 }
 
 // ReplaceIndexInDashboardObject replaces references to the index pattern in dashboard objects
-func ReplaceIndexInDashboardObject(index string, content common.MapStr) common.MapStr {
+func ReplaceIndexInDashboardObject(index string, content []byte) []byte {
 	logger := logp.NewLogger("dashboards")
 	if index == "" {
 		return content
 	}
 
-	objects, ok := content["objects"].([]interface{})
-	if !ok {
-		return content
-	}
-
-	for i, object := range objects {
-		objectMap, ok := object.(map[string]interface{})
-		if !ok {
+	result := make([]byte, 0)
+	r := bufio.NewReader(bytes.NewReader(content))
+	for {
+		line, err := r.ReadBytes('\n')
+		if err != nil {
+			if err == io.EOF {
```

Comment:
I think there's a subtle usage issue here in that `ReadBytes` is documented to return both data and io.EOF in some cases. So it should handle that data that is returned in this case. I recommend using `bufio.Scanner` here if you can (seems easier to use).

> If ReadBytes encounters an error before finding a delimiter, it returns the data read before the error and the error itself (often io.EOF).

Reply:
I initially started with `bufio.Scanner`, but the tokens were too big so I switched to `bufio.Reader` as the documentation recommended:

> Programs that need more control over error handling or large tokens, or must run sequential scans on a reader, should use bufio.Reader instead.

Reply:
Don't you think it there should be some kind of error returned or error handling for malformed data that does not end in a newline? This implementation silently discards the data.

Reply:
There an error is logged a few lines below when there is an error. Also, the original data remain untouched, so the user can rerun the command after inspecting/fixing the input data.

Reply:
Indeed. I have tested with incomplete lines in a more complex scenario where it was correct. I have looked at the data exported from Kibana also, the spec from http://ndjson.org/ and it seems that EOF is a valid separator as well. So the errors we have to handle is anything besides EOF. So I still believe this is correct. :D